docs/user_guide/datasets.rst

```diff
@@ -0,0 +1,442 @@
+    ... sensitive_features=X_test["B"])
+    >>> print(grouped_metric.by_group)
+           accuracy     precision       recall   false positive rate   true positive rate    selection rate  count
+    B                                                                                    
+    False  0.852459          0.92     0.766667              0.064516             0.766667          0.409836     61   
+    True   0.863636      0.909091     0.833333                   0.1             0.833333               0.5     66     
+
+    >>> grouped_metric = MetricFrame(metrics=metrics,
+    ... y_true=y_test,
+    ... y_pred=predicted,
+    ... sensitive_features=X_test["LSTAT"])
+    >>> print(grouped_metric.by_group)
+           accuracy     precision       recall   false positive rate   true positive rate    selection rate  count
+    LSTAT                                                                        
+    False  0.901639      0.910714     0.980769              0.555556             0.980769          0.918033     61 
+    True   0.818182           1.0     0.142857                   0.0             0.142857          0.030303     66 
+    
+
+.. _discussion:
+
+Discussion
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Boston housing dataset presents many ethical issues, and in general, we 
+strongly discourage using it in predictive modelling analyses. 
+We've kept it in Fairlearn because of its potential as a teaching tool 
+for how to deal with ethical issues in a dataset. 
+There are ways to remove correlations between sensitive features and the remaining columns [#7]_, 
+but that is by no means a guarantee that fairness-related harms won't occur. 
+Besides, other benchmark datasets exist that do not present these issues.
+
+
+It's important to keep the differences between the way Harrison and Rubenfield 
+used the dataset and the way modern machine learning practicioners have used 
+it in focus. 
+Harrison and Rubenfield conducted an empirical econometric study,
+the goal of which was to determine the causal impacts of these variables on 
+median home value. 
+Interpretation of causal models involves looking at model
+coefficients to ascertain the effect of one variable on the dependent variable,
+holding all other factors constant. This use case is different than the typical 
+supervised learning analysis. 
+A machine learning model will pick up on the 
+patterns encoded in the data and use that to predict an outcome.
+In the Boston housing dataset, the patterns the authors encoded through
+the *B* and *LSTAT* variables include systemic racism and class inequalities, 
+respectively. 
+Using the Boston housing dataset as a benchmark for a new 
+supervised learning model means that the model's performance is in part due to
+how well it learns and replicates the patterns in this dataset.
+
+The Boston Housing dataset raises the more general issue of whether it's valid to 
+port datasets constructed for one specific use case to different use cases.
```

Comment:
@mmadaio I committed an updated version of the Fairness and Machine Learning page where the only updates are refs to the various sections (in case someone else wants to reference that page in the future). Please let me know if it looks right!